packages/charting/src/components/StackedBarChart/StackedBarChart.base.tsx

```diff
@@ -127,12 +153,39 @@ export class StackedBarChartBase extends React.Component<IStackedBarChartProps,
     });
   }
 
+  private _getTotalChartData(data: IChartProps): number {
+    let sum = 0;
+    if (data.chartData && data.chartData.length > 0) {
+      const chartData = data.chartData;
+      for (let i = 0, n = chartData.length; i < n; i++) {
+        sum += chartData[i].data || 0;
+      }
+    }
+
+    return sum;
+  }
+
+  private _createBenchmark(totalData: number, dataPoint: IChartDataPoint): JSX.Element {
+    const ratio = (dataPoint.data! / totalData) * 100;
+
+    const styles = {
+      marginLeft: 'calc(' + ratio + '% - 4.5px)',
```

Comment:
@dzearing are we okay with calc() usage here?

Reply:
move the code `'calc(' + ratio + '% - 4.5px)'` into StackedBarChart.styles.ts

Reply:
1) yeah all styles should be in .styles.ts as much as possible, and things necessary to determine the styles, like ratio, should be passed in as styleProps
2) I was also asking about the perf of calc(), since it can be slow if it's used overly much or on items that get recalculated (such as for layout). I don't think it's a big issue here though so it should be fine.

Reply:
@phkuo yup agreed i put a similar comment in above. If we don't put it in styles then it also becomes extremely difficult to customize without doing silly things like manipulating the dom or using global styles.